services-core/src/economic_viability.rs

```diff
@@ -30,22 +32,29 @@ impl EconomicViabilityStrategy {
         fallback_max_gas_price: u128,
         native_token_price: Arc<dyn NativeTokenPricing + Send + Sync>,
         gas_station: Arc<dyn GasPriceEstimating + Send + Sync>,
-    ) -> impl EconomicViabilityComputing {
-        let mut economic_viabilities = Vec::<Box<dyn EconomicViabilityComputing>>::new();
-        if matches!(self, EconomicViabilityStrategy::Dynamic) {
-            economic_viabilities.push(Box::new(EconomicViabilityComputer::new(
-                native_token_price,
-                gas_station,
-                subsidy_factor,
-                min_avg_fee_factor,
-            )));
-        }
-        // This should come after the subsidy calculation so that it is only used when that fails.
-        economic_viabilities.push(Box::new(FixedEconomicViabilityComputer::new(
+    ) -> Arc<dyn EconomicViabilityComputing> {
+        let fixed = FixedEconomicViabilityComputer::new(
             Some(fallback_min_avg_fee_per_order),
             Some(fallback_max_gas_price.into()),
-        )));
-        PriorityEconomicViabilityComputer::new(economic_viabilities)
+        );
+        let dynamic = EconomicViabilityComputer::new(
+            native_token_price,
```

Comment:
I assume `native_token_price` is specified at runtime? Although I suppose it would be possible to fetch this price based on the existing orderbook for token 0, perhaps this would be too involved and not guaranteed to be deterministic.

Reply:
The logic of how we get native_token_price hasn't changed. From the perspective of this function it's just a trait so this function doesn't know how it's implemented. Iirc we use the usual token price prediction here which is third party price estimators and also the onchain orderbook.